internal/proxy/proxy.go

```diff
@@ -132,15 +139,16 @@ func (c *portConfig) nextDBPort(version string) int {
 
 // Client represents the state of the current instantiation of the proxy.
 type Client struct {
-	cmd    *cobra.Command
 	dialer cloudsql.Dialer
 
 	// mnts is a list of all mounted sockets for this client
 	mnts []*socketMount
+
+	logger log.Logger
```

Comment:
Shouldn't this say a reference to cmd? What if cmd's logger changes?

Reply:
I think our contract here is that prior to calling `Execute`, the logger can change, but after, the command has started and the configuration is set. Side note: Hot reloading will probably create a new command (saving some state from the previous command) and if the logger changes, it will still be done before the next call to `Execute`.

Reply:
I think we'll want hotloading to preserve the existing connections, so my 2c is that the cmd will need to be updated and adapted instead. But we can handle that in a different PR if need be.